### PR TITLE
challenger: add unit tests for `observe_base_as_algebra_element`

### PR DESCRIPTION
**Pull request by tcoratger**
Followup on the top of https://github.com/Plonky3/Plonky3/pull/1152

### CODE REVIEW

File: challenger/src/duplex_challenger.rs [hunk 1]
```diff
@@ -471,4 +474,133 @@ mod tests {
...
+
+        // Verify the buffer contains the correct basis coefficients
+        //
+        // First coefficient should be the original base field value
+        assert_eq!(chal.input_buffer[initial_len], G::from_u8(42));
+        // Second coefficient should be zero (since we're embedding base into extension)
+        assert_eq!(chal.input_buffer[initial_len + 1], G::ZERO);
+
+        // Output buffer should remain empty until we sample
+        assert!(chal.output_buffer.is_empty());
```
Comment (SyxtonPrime) [inline challenger/src/duplex_challenger.rs]: Hmm, this is hard coding in the expected map from `G` to `EF2G`. In general Algebra maps may not look like this, (This could also break if we change the basis we use for `EF2G`.

I think `test_observe_base_as_algebra_element_consistency_with_direct_observe` is the key test here. E.g. we just need to ensure that `observe_base_as_algebra_element` is identical to first embedding into the larger algebra and then observing.

TBH unless there is an edge case which wouldn't be picked up by that test, I'd be in favor of keeping that one and getting rid of the rest.

File: challenger/src/duplex_challenger.rs [hunk 2]
```diff
@@ -471,4 +474,133 @@ mod tests {
...
+
+        // Method 2: Manually convert and observe via observe_algebra_element
+        let ext_val = EF2G::from(base_val);
+        chal2.observe_algebra_element(ext_val);
+
+        // Both methods should produce identical internal state
+        assert_eq!(chal1.input_buffer, chal2.input_buffer);
+        assert_eq!(chal1.output_buffer, chal2.output_buffer);
+        assert_eq!(chal1.sponge_state, chal2.sponge_state);
+    }
```
Comment (SyxtonPrime) [inline challenger/src/duplex_challenger.rs:499]: Reiterating above, I'd vote to have this test and one more test similar to this one but where we observe a stream of base field elements and check that the internal states all match up.

File: challenger/src/duplex_challenger.rs (hunk 1 above)
Comment (tcoratger) [reply challenger/src/duplex_challenger.rs]: Ok no problem that makes sense, just fixed this to keep only 2 tests with line by line doc to explain what we are doing.

File: challenger/src/duplex_challenger.rs (hunk 2 above)
Comment (tcoratger) [reply challenger/src/duplex_challenger.rs:499]: Fixed

**Comment by tcoratger** [comment]
@SyxtonPrime Let me know if I can do anything on this one? I feel we are good to merge no?
